game-core/src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1528,6 +1528,15 @@ public static Route getBestRoute(final Territory start, final Territory end, fin
         notEnemyOwned.and(noEnemyUnits),
         noEnemyUnits,
         noAa));
+
+    // Determine max distance route is willing to accept
+    final List<Unit> landUnits =
+        CollectionUtils.getMatches(unitsWhichAreNotBeingTransportedOrDependent, Matches.unitIsLand());
+    final int maxLandMoves = landUnits.isEmpty() ? 0 : getMaxMovement(landUnits);
+    final int maxSteps = GameStepPropertiesHelper.isCombatMove(data) ? defaultRoute.numberOfSteps()
+        : Math.max(defaultRoute.numberOfSteps(), maxLandMoves);
```

Comment:
shouldn't this be Math.min ?

Reply:
Nevermind, I got it

Reply:
Nope. If max land moves is larger than the default route than I want to take that as the new distance to try for. Otherwise I will just use the default route distance.